conjure-undertow-processor/src/main/java/com/palantir/conjure/java/undertow/processor/data/ParamTypesResolver.java

```diff
@@ -349,6 +354,32 @@ private CodeBlock getCollectionParamDecoder(
                 });
     }
 
+    private Optional<CodeBlock> getParamDecoderConstructor(DeclaredType declaredType) {
+        TypeElement typeElement = (TypeElement) declaredType.asElement();
+        // new T(String)
+        return typeElement.getEnclosedElements().stream()
+                .filter(element -> element.getKind() == ElementKind.CONSTRUCTOR)
```

Comment:
possibly worth excluding private elements here

Reply:
Ah, yes.

The checks in `getUnknownDecoderFactoryFunction` require public constructors/methods. I originally had that here and then the tests failed because I didn't use public constructors in my test cases. I figured that was unnecessarily restrictive and removed it.

I think this in consistency is desirable. There's probably a meaningful difference between a factory method used to instantiate a value and a factory methods used to instantiate a decoder. We probably want to require public elements for the former but it's okay to allow package private elements for the later.

As part of this it might be good to add a similar check to `Instantiables`. Right now it throws if there are multiple enum values and generated invalid code if there is not a no-args constructor. At the very least we should add a check, although it would be better to refactor this a bit and report errors. I'll plan to do that in a FLUP.